File: OCH_Tao/ABC/B/370.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
let A = [];
for (let i = 1; i < N + 1; i++) {
  A.push(input[i].split(" ").map(x => parseInt(x, 10)));
}
let ans = 1;
for (let i = 0; i < N; i++) {
  let tmp = [ans - 1, i];
  ans = A[Math.max(...tmp)][Math.min(...tmp)];
}
console.log(ans);

File: OCH_Tao/ABC/B/370.py
N = int(input())
A = []
for i in range(N):
  A.append(list(map(int,input().split())))
ans = 1
for i in range(N):
  tmp = [ans-1,i]
  ans = A[max(tmp)][min(tmp)]
print(ans)